wallet-api/src/Wallet/Emulator/Http.hs

```diff
@@ -51,7 +51,7 @@ type WalletAPI
      :<|> "wallets" :> Capture "walletid" Wallet :> "payments" :> ReqBody '[ JSON] Value :> Post '[ JSON] ( Set TxIn'
                                                                                                           , TxOut')
      :<|> "wallets" :> Capture "walletid" Wallet :> "pay-to-public-key" :> ReqBody '[ JSON] Value :> Post '[ JSON] TxOut'
-     :<|> "wallets" :> Capture "walletid" Wallet :> "transactions" :> ReqBody '[ JSON] Tx :> Post '[ JSON] ()
+     :<|> "wallets" :> Capture "walletid" Wallet :> "transactions" :> ReqBody '[ JSON] Tx :> Post '[ JSON] [Tx]
```

Comment:
This is where the line between wallet API and control API is crossed (returning the `[Tx]` only makes sense when running a `WalletAPI m => m ()` inside a `Trace`, but not on the wallet API on its own (otherwise the signature of `submitTxn` would be `submitTxn :: Tx -> m [Tx]`)). 

Unfortunately we need to return the `Tx` here because we have to reference it later. So I can't see a way around this change.

Reply:
done